src/Featurizers/Components/GrainFeaturizerImpl.h

```diff
@@ -215,11 +213,9 @@ class GrainEstimatorImplBase : public BaseT {
     // |
     // ----------------------------------------------------------------------
     using EstimatorMap =
-        std::unordered_map<
+        std::map<
             GrainT,
-            EstimatorT,
-            std::hash<GrainT>,

```

Comment:
not a comment. 
if using unordered_map, hash function should be customly defined, std::hash<T> probably not gonna work

Reply:
You just need to define a specialization for your type for std::hash<>, that is how  it is used.

---
In reply to: [389183880](https://github.com/microsoft/FeaturizersLibrary/pull/160#discussion_r389183880) [](ancestors = 389183880)